### PR TITLE
Clarify fixed length types

### PR DESCRIPTION
**Pull request by pataei**
This PR clarifies the use of `bit<W>`, `int<W>`, and `varbit<W>`. More specifically, it states that these represent the case where the width of the type is either a constant integer or a compile-time known expression that evaluates to an integer. In other words, it addresses issue #1228.

### CODE REVIEW

File: p4-16/spec/P4-16-spec.mdk
Comment (pataei) [inline p4-16/spec/P4-16-spec.mdk:4214]: This condition is only correct for signed integers as the underlying type of an enum and not for unsigned integers. It might be better to discuss when these types are well-formed (the conditions of their width holds) and state that when a type is used anywhere it's assumed that it is well-formed so we don't have to keep repeating these conditions.

**Review by mbudiu-vmw — APPROVED** [review]
This change also looks like could be merged without an LDWG discussion.
But I will wait for at least one more review.

**Review by jnfoster — APPROVED** [review]
Agree. LGTM!

**Pull request merged**
